### PR TITLE
New task in layouts.py - Add Lightning Action to Page Layouts

### PR DESCRIPTION
**Pull request by ClayTomerlin**
Adding new `metadata_etl` task that allows for addition of named Lightning Action/Button to a Page Layout.
- Use case example: Adding an unpackaged Quick Action to a packaged page layout's list of Lightning Actions. 

Reference for PlatformActionList metadata:
https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#PlatformActionList

# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

**Comment by jstvz** [comment]
@ClayTomerlin Moving this PR to ready to review to match the WI.

File: cumulusci/tasks/metadata_etl/tests/test_layouts.py [hunk 1]
```diff
@@ -177,3 +180,331 @@ def test_skips_existing_related_list(self):
...
+        )
+
+        # Creating expected action item/list xml and metadata
+        #   Expected context = "Record"
+        #   our action_list_items <sortOrder> is being set dynamically from the mock_action_list_size (if we need to change later)
+        #       setting our expected new action item to sortOrder 0 since placement should be first
+        #   using our optional_first_action_item to set our expected_action_item
+        expected_action_item = EMPTY_ACTION_ITEM.format(expected_order=0, **options)
+        expected_action_list = MOCK_EXISTING_ACTION_LIST.format(
+            *tuple([n for n in range(1, mock_action_list_size + 1)]),
```
Comment (jstvz) [inline cumulusci/tasks/metadata_etl/tests/test_layouts.py:373]:
```suggestion
            *range(1, mock_action_list_size + 1),
```

File: cumulusci/tasks/metadata_etl/tests/test_layouts.py [hunk 2]
```diff
@@ -177,3 +180,331 @@ def test_skips_existing_related_list(self):
...
+            )
+        )
+
+        # run test actual
+        actual_metadata = task._transform_entity(metadata, "Layout")
+        # Assert our transformed metadata is the same as our expected
+        # This confirms, action list item size, sortOrder, and record context
+        assert (
+            actual_metadata.tostring() == expected_metadata.tostring()
+        ), "The transformed actual_metadata should match our expected_metadata, with the new action item placed first"
```
Comment (jstvz) [inline cumulusci/tasks/metadata_etl/tests/test_layouts.py:390]: I recommend moving the message from the assertion to the comment above.

File: cumulusci/tasks/metadata_etl/tests/test_layouts.py [hunk 3]
```diff
@@ -177,3 +180,331 @@ def test_skips_existing_related_list(self):
...
+                "utf-8"
+            )
+        )
+
+        # run test
+        actual_metadata = task._transform_entity(metadata, "Layout")
+        # should not transform, and metadata should be none
+        assert (
+            actual_metadata is None
+        ), "AddRecordPlatformActionListItem should return None when actionListItem already exists in metadata"
```
Comment (jstvz) [inline cumulusci/tasks/metadata_etl/tests/test_layouts.py:423]: I recommend moving this message from the assertion to the comment above.

File: cumulusci/tasks/metadata_etl/tests/test_layouts.py [hunk 4]
```diff
@@ -177,3 +180,331 @@ def test_skips_existing_related_list(self):
...
+
+        # Creating expected action item/list xml and metadata
+        #   Expected context = "Record"
+        #   The action_list_items <sortOrder> (positional *args) can be dynamically set from the mock_action_list_size or range
+        #   using our optional_last_action_item to set our expected_action_item placement
+        expected_action_item = EMPTY_ACTION_ITEM.format(
+            expected_order=mock_action_list_size, **options
+        )
+        expected_action_list = MOCK_EXISTING_ACTION_LIST.format(
+            *tuple([n for n in range(mock_action_list_size)]),
```
Comment (jstvz) [inline cumulusci/tasks/metadata_etl/tests/test_layouts.py:316]: `range` is also a sequence, so you can unpack it here:
```suggestion
            *range(1, mock_action_list_size + 1),
```

File: cumulusci/tasks/metadata_etl/layouts.py [hunk 5]
```diff
@@ -66,5 +65,102 @@ def _create_new_related_list(self, metadata, api_name, related_list):
...
+        super()._init_options(kwargs)
+
+        self._action_type = self.options.get("action_type", "")
+        self._action_name = self._inject_namespace(self.options.get("action_name", ""))
+        # Default to False if `place_first` option was not set
+        self._place_first = (
+            False
+            if not self.options.get("place_first")
+            else process_bool_arg(self.options.get("place_first"))
+        )
```
Comment (jstvz) [inline cumulusci/tasks/metadata_etl/layouts.py:109]: Consider instead:

```suggestion
        self._place_first = process_bool_arg(
            self.options.get("place_first") or False
        )
```

File: cumulusci/tasks/metadata_etl/layouts.py [hunk 6]
```diff
@@ -66,5 +65,102 @@ def _create_new_related_list(self, metadata, api_name, related_list):
...
+        "action_type": {
+            "description": "platformActionListItems.actionType like 'QuickAction' or 'CustomButton'  See documentation: https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#PlatformActionListItem",  # noqa: E501
+            "required": True,
+        },
+        "action_name": {
+            "description": "platformActionListItems.actionName. The API name for the action to be added.  See documentation: https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#PlatformActionListItem",  # noqa: E501
+            "required": True,
+        },
+        "place_first": {
+            "description": "When 'True' the specified Record Platform Action List Item will be inserted before any existing on the layout. Default is 'False'",
```
Comment (jstvz) [inline cumulusci/tasks/metadata_etl/layouts.py:93]: I would tweak this to be consistent with the other options:
```suggestion
            "description": "When 'True' the specified Record platformActionListItem will be inserted before any existing on the layout. Default is 'False'",
```

**Review by jstvz — CHANGES_REQUESTED** [review]
Thanks! I've requested a few tweaks.

File: cumulusci/tasks/metadata_etl/tests/test_layouts.py (hunk 1 above)
Comment (ClayTomerlin) [reply cumulusci/tasks/metadata_etl/tests/test_layouts.py:373]: Ah, I was overthinking it a bit here

**Comment by ClayTomerlin** [comment]
@jstvz  Changes complete - thank you for the guidance and tips, let me know if I missed anything

**Review by jstvz — APPROVED** [review]

**Pull request merged**